app/Services/MapboxGeocodingService.php

```diff
@@ -9,21 +9,53 @@
 
 class MapboxGeocodingService
 {
-    public function getCoordinatesFromApi(string $cityName, string $countryName): array
+    protected string $city;
+    protected string $country;
+
+    public function getCoordinatesFromApi(string $cityName, string $countryName, $client = new Client()): array
     {
-        $token = env("MAPBOX_TOKEN");
-        $client = new Client();
+        $token = config("app.mapbox_token");
 
         try {
             $response = $client->get(
-                "https://api.mapbox.com/geocoding/v5/mapbox.places/$cityName,$countryName.json?access_token=$token&types=place",
+                config("app.mapbox_api_url") . "$cityName,$countryName.json?access_token=$token&types=place",
             );
 
-            $coordinates = json_decode($response->getBody()->getContents(), true)["features"][0]["center"];
+            $coordinates = json_decode($response->getBody()->getContents(), associative: true)["features"][0]["center"];
 
             return [$coordinates[1], $coordinates[0]];
         } catch (Throwable) {
             return [];
         }
     }
+
+    public function getPlaceFromApi(string $lat, string $long, $client = new Client()): array
```

Comment:
I mean to move `Client` to the `__constructor()`. Then This service class will need to be injected to in constructors.
This will allow you to write tests for this MapboxGeocoding service.